### PR TITLE
fixing type deprecation

### PR DESCRIPTION
**Pull request by kunal-iitkgp**
ping @isuruf @certik
executed the following
`julia -e 'Pkg.build("SymEngine")'`
 `julia -e 'Pkg.test("SymEngine")'` 
 still getting the same warnings

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,3 +1,3 @@
-julia 0.3
+julia 0.4
```
Comment (isuruf) [inline REQUIRE]: Is julia 0.3 not supported anymore?

Comment (kunal-iitkgp) [inline REQUIRE]: Julia 0.4 is a stable version so we can use it.

**Comment by isuruf** [comment]
Changes have been made to symengine master since the last commit. Can you fix the CI issues?

**Comment by kunal-iitkgp** [comment]
@isuruf For the CI issues I think g++ 4.7 should be added in addon in `travis.yml`? I will try to fix this
And can you pls clarify by "Changes have been made in symengine master since the last commit "

**Comment by isuruf** [comment]
Yes, I meant the same

**Comment by kunal-iitkgp** [comment]
I think this will PR will fix the type depreciation warnings if #12 gets merged and also I want to send a PR which wraps `ntheory` 
ping @isuruf

**Comment by isuruf** [comment]
Sure. Before wrapping ntheory, I think it's important to implement a type system in Julia

**Comment by kunal-iitkgp** [comment]
Do you mean mapping Symengine types to Julia ? @isuruf

**Comment by isuruf** [comment]
Yes. I'm going to reopen this PR to trigger a new build

**Comment by kunal-iitkgp** [comment]
@isuruf fixed all the warnings

**Comment by isuruf** [comment]
Thanks for the PR

**Pull request merged**